os_migrate/plugins/module_utils/reference.py

```diff
@@ -374,6 +374,35 @@ def user_ref(conn, id_, required=True, allow_auth=True):
     return ref
 
 
+def role_id(conn, ref, required=True):
+    """Fetch ID of a Role identified by reference dict `ref`. Use
+    OpenStack SDK connection `conn` to fetch the info. If `required`,
+    ensure the fetch is successful.
+    Returns: the ID, or None if not found and not `required`
+    Raises: openstack's ResourceNotFound when `required` but not found
+    """
+    return _fetch_id(conn, conn.identity.find_role, ref, required)
+
+
+def role_ref(conn, id_, required=True):
+    """Create reference dict for a Role identified by ID `id_`. Use
+    OpenStack SDK connection `conn` to fetch the info. If `required`,
+    ensure the fetch is successful.
+    Returns: the ref dict, or None if not found and not `required`
+    Raises: openstack's ResourceNotFound when `required` but not found
+    """
+    ref = _fetch_ref(conn, conn.identity.find_role, id_, required)
```

Comment:
Since roles don't belong to projects, we'll need `get_project_info=False` here too, see definition of user_ref and project_ref.